aiida/manage/database/delete/nodes.py

```diff
@@ -58,43 +80,70 @@ def delete_nodes(
         else:
             starting_pks.append(pk)
 
+    # An empty set might be problematic for the queries done below.
     if not starting_pks:
-        # I prefer checking explicitly, an empty set might be problematic for the queries done below.
         if verbosity:
             echo.echo('Nothing to delete')
         return
 
-    # The following code is just for the querying of downwards provenance.
-    # Ideally, there should be a module to interface with, but this is the solution for now.
-    # By only dealing with ids, and keeping track of what has been already
-    # visited in the query, there's good performance and no infinite loops.
-    link_types_to_follow = [LinkType.CREATE.value, LinkType.INPUT_CALC.value, LinkType.INPUT_WORK.value]
-    if follow_calls:
-        link_types_to_follow.append(LinkType.CALL_CALC.value)
-        link_types_to_follow.append(LinkType.CALL_WORK.value)
-    if follow_returns:
-        link_types_to_follow.append(LinkType.RETURN.value)
-
-    edge_filters = {'type': {'in': link_types_to_follow}}
-
-    # Operational set always includes the recently (in the last iteration added) nodes.
-    operational_set = set().union(set(starting_pks))  # Union to copy the set!
-    pks_set_to_delete = set().union(set(starting_pks))
+    follow_backwards = []
+    follow_backwards.append(LinkType.CREATE.value)
+    follow_backwards.append(LinkType.RETURN.value)
+    follow_backwards.append(LinkType.CALL_CALC.value)
+    follow_backwards.append(LinkType.CALL_WORK.value)
+
+    follow_fowards = []
```

Comment:
Sorry to bother, but it should be `forwards` and not `fowards`...

Reply:
Uf, aplogies, my bad!